src/core/executors/ExecutorResult.ts

```diff
@@ -1,7 +1,29 @@
-export default class ExecutorResult<Model> {
-  constructor(
-    readonly success: boolean,
-    readonly retriable: boolean,
-    readonly result?: Model
-    ){}
+export default interface ExecutorResult<Model> {
```

Comment:
no longer need to export

Reply:
This still needs to be exported as it is still a result type used outside of this file.
But it should probably don't be `default`, so I can fix that.

Reply:
Removed `default` here